### ISSUE TITLE
Add `Trainer.info()`

### ISSUE THREAD

**Issue by arxyzan** [hezarai/hezar#120]
When a user wants to train a model using the `Trainer`, it just starts with no further info about the training properties. It's better to provide a function in the `Trainer ` to generate a user-friendly info message containing all the necessary propreties of the trainer. 
The properties I have in mind right now are:
- Model type
- Device(s)
- Dataset
- Optimizer
- Initial learning rate
- Total samples
- Epochs
- Batch size
- Total parameters
- Trainable parameters
- Mixed precision
- Metrics
- Checkpoints path

The output is something like below:
```
******************** Training Info ********************

Model type: T5TextGeneration
Device(s): ['cuda']
Dataset: `hezarai/xlsum-fa` (text_summarization)
Optimizer: `adam`
Initial learning rate: ...
Total samples: ...
Epochs: ...
Batch size: ...
Total parameters: ...
Trainable parameters: ...
Mixed precision: True
Metrics: ['rouge']
Checkpoints path: ...

*****************************************************

```

To wrap up, the task is to add a new method to the `Trainer` class called `info()` that just prints the above properties. Next, this method should be called in `Trainer.train()` before everything.
